torch/serialization.py

```diff
@@ -203,16 +203,39 @@ def persistent_id(obj):
 def load(f, map_location=None, pickle_module=pickle):
     """Loads an object saved with :func:`torch.save` from a file.
 
-    torch.load can dynamically remap storages to be loaded on a different device
-    using the map_location argument. If it's a callable, it will be called with
-    two arguments: storage and location tag. It's expected to either return a
-    storage that's been moved to a different location, or None (and the location
-    will be resolved using the default method). If this argument is a dict it's
-    expected to be a mapping from location tags used in a file, to location
-    tags of the current system.
-
-    By default the location tags are 'cpu' for host tensors and 'cuda:device_id'
-    (e.g. 'cuda:2') for cuda tensors. User extensions can register their own
+    torch.load uses python's normal unpickling facilities but treats storages,
+    which underlie tensors, specially.  Storages are first deserialized on the
+    cpu.  Then, by default, torch.load tries to move the storage to the device
+    it was saved from, raising an exception if this fails because the device
+    does not exist.  If you serialize a cuda tensor with torch.save, when you
+    deserialize it with torch.load, it will be created on the device it was
+    originally on from a storage that has been moved there.
```

Comment:
Don't the last two sentences say exactly the same thing?

Reply:
I can delete the last sentence if you prefer, but the point was that most users think about tensors while the mechanism actually operates on storages.  So the last sentence sums up what the upshot is if you stored a tensor.  Maybe people can draw that inference themselves though if brevity is key.

Reply:
Yes, I'd prefer to keep it short. How about "torch.load tries to move storages and tensors to the device it was saved from, ..."